chapter_recurrent-modern/machine-translation-and-dataset.md

```diff
@@ -118,13 +131,27 @@ def build_array(lines, vocab, num_steps, is_source):
     return array, valid_len
 ```
 
+```{.python .input}
+#@tab pytorch
+#@save
+def build_array(lines, vocab, num_steps, is_source):
+    lines = [vocab[l] for l in lines]
+    if not is_source:
+        lines = [[vocab['<bos>']] + l + [vocab['<eos>']] for l in lines]
+    array = torch.tensor([truncate_pad(
+        l, num_steps, vocab['<pad>']) for l in lines])
+    valid_len = (array != vocab['<pad>']).sum(axis=1)
```

Comment:
```suggestion
    valid_len = (array != vocab['<pad>']).sum(dim=1)
```
Let's use `dim` instead of `axis` just like the PyTorch documentation which uses `dim` almost everywhere.